controller/src/main/java/io/pravega/controller/server/rpc/grpc/v1/ControllerServiceImpl.java

```diff
@@ -477,19 +485,29 @@ public void getDelegationToken(StreamInfo request, StreamObserver<DelegationToke
         authenticateExecuteAndProcessResults(() -> this.grpcAuthHelper.checkAuthorizationAndCreateToken(
                 AuthResourceRepresentation.ofStreamInScope(request.getScope(), request.getStream()),
                 AuthHandler.Permissions.READ_UPDATE),
-                delegationToken -> CompletableFuture.completedFuture(Controller.DelegationToken
-                        .newBuilder()
-                        .setDelegationToken(delegationToken)
-                        .build()),
+                delegationToken -> {
+                    logIfEmpty(delegationToken, "getDelegationToken", request.getScope(), request.getStream());
+                    return CompletableFuture.completedFuture(DelegationToken
+                            .newBuilder()
+                            .setDelegationToken(delegationToken)
+                            .build());
+                },
                 responseObserver);
     }
 
+    private void logIfEmpty(String delegationToken, String requestName, String scopeName,
+                            String streamName) {
+        if (isAuthEnabled() && (delegationToken == null || delegationToken.equals(""))) {
```

Comment:
we can use guava's Strings.isNullOrEmpty

Reply:
Sure, that sounds good. I've modified the code and pushed the change.